### ISSUE TITLE
[QUESTION] Forcing TFT model predictions to start at zero

### ISSUE THREAD

**Issue by benjamin-strath** [unit8co/darts#2602]
Hi there,

I was wondering if anyone knew of a way to force predictions made using the TFT model to start at zero. 
I ask because all of the time series I am using to train and test my model start at 0 but when making predictions the starting value is slightly off. 

Any help would be greatly appreciated.

**Comment by madtoinou** [comment]
Hi @benjamin-strath,

To my knowledge, there is no elegant way to force a zero value at the first timestamp of the predictions directly in the model. You could eventually have a post-processing pipeline to shift all the forecasts so that it always start at 0 but by doing so, you might increase the error at other timestamps.

Note that the first timestamp of each series is never part of the predicted values (because there is no values before to be used as features), so unless there is some periodicity in your data and you always make it so that the validation/test set start at the beginning of a cycle (at 0), I am not sure to understand what you mean by "all the time-series [...] start at 0".

**Comment by madtoinou** [comment]
Forcing is a big word, I would say that it's more about tricking/helping the mode to learn this pattern. You could try to create an future covariates, which could for example encode the distance to the beginning of the forecasts or maybe just mark the beginning of the forecasts (but 0 the rest of the time). Of course, this must be done for the series included in the training set as well.

However, I would say that such manipulation is not recommended and having a slight offset for this position should not be a big deal. I you really want this value to be 0, relying on reconciliation/post-processing to correct this kind of deviation should definitely be favored.
